app/components/share-controls.js

```diff
@@ -0,0 +1,30 @@
+import Component from '@ember/component';
+import { action } from '@ember-decorators/object';
+import { argument } from '@ember-decorators/argument';
+import { run } from '@ember/runloop';
+
+export default class ShareControls extends Component {
+  @argument shareURL;
+  @argument closed = true;
+  @argument copySuccess = false;
+
+  @action
+  handleShareOpen() {
+    this.set('closed', false);
+  }
+
+  @action
+  handleShareClose() {
+    this.set('closed', true);
+    this.set('copySuccess', false);
+  }
+
+  @action
+  handleShareSuccess() {
+    this.set('copySuccess', true);
+
+    run.later(this, function() {
+      this.set('copySuccess', false);
+    }, 2000);
```

Comment:
You can also say:

```javascript
run.later(() => {
  this.set('copySuccess', false);
});
```

Arrow functions replace their `this` context with their outer lexical scope.